app/scripts/lib/multichain-api/provider-authorize/handler.js

```diff
@@ -64,7 +72,45 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
     );
   }
 
+  const networkClientIdsAdded = [];
+
   try {
+    const { flattenedRequiredScopes, flattenedOptionalScopes } = processScopes(
+      requiredScopes,
+      optionalScopes,
+    );
+
+    const validScopedProperties = processScopedProperties(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
+      scopedProperties,
+    );
+
+    const existsNetworkClientOrEip3085ForChainId = (chainId) => {
+      const scopeString = `eip155:${parseInt(chainId, 16)}`;
+      if (validScopedProperties?.[scopeString]?.eip3085) {
+        return true;
+      }
+      try {
+        findNetworkClientIdByChainId(chainId);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    };
+
+    const supportedRequiredScopes = flattenedRequiredScopes;
+    assertScopesSupported(flattenedRequiredScopes, {
+      existsNetworkClientForChainId: existsNetworkClientOrEip3085ForChainId,
+    });
+
+    const supportedOptionalScopes = filterScopesSupported(
+      flattenedOptionalScopes,
+      {
+        existsNetworkClientForChainId: existsNetworkClientOrEip3085ForChainId,
```

Comment:
any reason to not just rename this hook too?

Reply:
it becomes overly specific. filterScopesSupported can still be used to filter out scopes that aren't _currently_ supported rather than filtering out scopes that _can be_ supported

Reply:
perhaps the hook should be called `isChainIdSupported`?

Reply:
This is kindof making me think these should be two different buckets. `supportedOptionalScopes` and `supportableOptionalScopes` or `optionalScopesToAdd`. Same with `requiredScopes`. These buckets are useful to send along to the approval confirmation.

Reply:
done starting from this commit 

@jiexi
[change from existsNetworkClientForChainId to isChainIdSupported](https://github.com/MetaMask/metamask-extension/pull/25873/commits/efd5cdbe3cdd4385b6e63f82d12bdee2a23dbc41)